File: Excel/707 Challenge.py
import pandas as pd
import re

path = "707 Split at Non Alphabetic Delimiters.xlsx"
input = pd.read_excel(path, usecols="A", nrows=10)
test = pd.read_excel(path, usecols="B", nrows=36)

result = input['Sentences'].str.split(r'[^a-zA-Z]+', expand=True).stack().reset_index(drop=True)

print(result.equals(test['Expected Answer'])) # True 

File: Excel/700-799/707/707 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/707 Split at Non Alphabetic Delimiters.xlsx"
input = read_excel(path, range = "A1:A10")
test = read_excel(path, range = "B1:B36")

result = input %>%
  separate_rows(Sentences, sep = "[^[:alpha:]]+")

all.equal(result$Sentences, test$`Expected Answer`)
#> [1] TRUE
